### ISSUE TITLE
Microtasks bypass benchmark?

### ISSUE THREAD

**Issue by stagas** [krausest/js-framework-benchmark#962]
I'm trying to figure out where the benchmark result is being validated because I have the sense that frameworks utilizing microtasks (using queueMicrotask/Promise/setTimeout) are delegating most of the work at the end of the event queue and after the benchmark has done its pass and reported back, which would completely skew all results in favor of those. Is this being handled somewhere?

**Comment by krausest** [comment]
I hope that the way we measure the duration avoids that. If we measured only JS script duration for the click event we'd completely run into this issue, but this is not how it works.
Instead we're performing the click event, wait until the expected dom mutation can be observed and then extract from chrome's performance trace the timestamp of the click event and the last observed paint event, no matter if there were pauses in between the execution. Thus I think there should be no problem. 

Which framework should we take a closer look at?

**Comment by stagas** [comment]
I see, then it shouldn't be an issue. I just wanted to check how the results are being measured because on my own benchmarks I noticed this behavior and until I took that into account I was getting skewed results so I was curious if this is happening here as well. But since you're saying you're not relying on the JS event queue reporting back then it shouldn't suffer from the same issue. Thanks for taking the time, I'll try to install this repo and do some more tests and will report back here if I find anything. Closing it for now.
